code/parse/sexp.cpp

```diff
@@ -12403,6 +12410,50 @@ void sexp_set_player_orders(int n)
 	}
 }
 
+void sexp_set_order_allowed_target(int n)
+{
+	bool allow_order;
+	std::set<size_t> orders;
+
+	auto ship_entry = eval_ship(n);
+	if (!ship_entry || !ship_entry->shipp) {
+		return;
+	}
+	auto shipp = ship_entry->shipp;
+
+	const std::set<size_t>& default_orders = ship_set_default_orders_against();
+	n = CDR(n);
+	allow_order = is_sexp_true(n);
+	n = CDR(n);
+	do {
+		for( size_t order : default_orders){
+			//Once we exceed the number of valid orders, break and warn
+			if (order >= Player_orders.size()) {
+				Warning(LOCATION, "Invalid order name %s found in sexp!", CTEXT(n));
+				break;
+			}
+			// OPF_AI_ORDER returns hud_name, so we must compare to that instead of parse_name
+			if (!stricmp(CTEXT(n), Player_orders[order].hud_name.c_str())) {
+				orders.insert(order);
+				break;
+			}
+		}
+
+		n = CDR(n);
+	}while (n >= 0);
+		
+	// set or unset the orders
+	if (allow_order) {
+		shipp->orders_allowed_against.insert(orders.begin(), orders.end());
+	}
+	else {
+		std::set<size_t> diff;
+		std::set_difference(shipp->orders_allowed_against.begin(), shipp->orders_allowed_against.end(), orders.begin(), orders.end(),
+							std::inserter(diff, diff.end()));
+		shipp->orders_allowed_against = diff;
```

Comment:
You're a pretty thorough tester, but I'm still going to ask if this logic branch was tested.  Not that I think you made any mistakes, but I've never used `set_difference` myself.  I probably would have wrote manual code (not that it's better, I just didn't know it existed).

Reply:
Yeah, I tested many variations of adding one or multiple orders and then removing one or multiple orders. It all seemed to work as expected.

Now, I don't pretend to fully comprehend Sets (seems a lot like a bitfield but for an object array). This was the first time I dealt with one, but most order lists in the game are done with Sets so I matched that. In fact almost this entire method was copied from the one above `sexp_set_player_orders`, including `set_difference`. I'm about as confident in the code as a novice coder like me can be.